src/shared/models/currencies.model.ts

```diff
@@ -1,6 +1,16 @@
-import { LiteralUnion } from '@shared/utils/type-utils';
+const CRYPTO_CURRENCIES_ARRAY = ['BTC', 'STX'] as const;
 
-export type CryptoCurrencies = LiteralUnion<'BTC' | 'STX', string>;
+export type CryptoCurrencies = (typeof CRYPTO_CURRENCIES_ARRAY)[number];
+
+export const isCryptoCurrency = (value: unknown): value is CryptoCurrencies =>
+  CRYPTO_CURRENCIES_ARRAY.some(cryptocurrency => cryptocurrency === value);
+
+export function convertValueToCryptocurrency(value: unknown) {
```

Comment:
Nit here, but just pointing out you sometimes camelCase `CryptoCurrency` but other times not.

Reply:
I vote Cryptocurrency as one word

Reply:
Thank you for catching this!